src/util/createCampaignQueryString.ts

```diff
@@ -1,26 +1,20 @@
-const disallowList = [
-	'addressType',
-	'amount',
-	'interval',
-	'paymentType',
-	'token',
-	'accessToken',
-	'utoken',
-	'id',
-	'addressToken',
-	'donationId',
-	'donationAccessToken',
-	'banner_submission',
-];
-
 export const QUERY_STRING_INJECTION_KEY = 'campaignQueryString';
 
-export function createCampaignQueryString( query: string ): string {
-	const searchParams = new URLSearchParams( query );
+/**
+ * Filter URL query parameters to return a URL query parameter that only contains allowed parameters and their values from the URL
+ *
+ * Example query string: spenden.wikimedia.de/?piwik_kw=123&id=5&utoken=badcafee&des=1&at=0
+ */
+export function createCampaignQueryString( query: string, allowedCampaignParameters: string[] ): string {
```

Comment:
I think we need to also allow the tracking parameters here? Eg `piwik_campaign`, `piwik_kwd`, `impCount`, `bImpCount`